packages/dataviews/src/field-types/boolean.tsx

```diff
@@ -65,7 +69,16 @@ export default {
 	render,
 	Edit: 'checkbox',
 	sort,
-	isValid,
+	getIsValid: ( field: Field< any > ) => ( {
+		required: isValidRequiredForBool( field.isValid?.required ),
+		min: false,
+		max: false,
+		pattern: false,
+		minLength: false,
+		maxLength: false,
+		elements: isValidElements( field.isValid?.elements ?? true ),
+		custom: field.isValid?.custom ?? isValidCustom,
+	} ),
```

Comment:
There's something weird in the architecture, that is visible here. Why should `getIsValid` of "boolean" care about returning minLenght` or `maxLenght` or `pattern` or something like that?

For me, the way I see this written is something like:

```js
combineValidation(
   isValidRequiredForBool,
   isValidElements,
   customValidationFunction
)
```

basically getIsValid would be of type `field: Field =>  value: T => Validity`

and 

you can combine multiple functions together, where each of this function has the same type `field: Field =>  value: T => Validity`

Reply:
This is about how we define the `NormalizedRules< Item >` type. 

This is the current type:

```ts
required:
		| false
		| {
				constraint: any;
				validate: (
					item: Item,
					field: NormalizedField< Item >
				) => boolean;
		  };
```

What you're proposing is that we make this change (any rule can be undefined):

```ts
required?: {
				constraint: any;
				validate: (
					item: Item,
					field: NormalizedField< Item >
				) => boolean;
		  };
```

I tend to prefer being explicit over implicit, and so I like the current approach more. While I have a preference, this is not something fundamental, to be honest. Happy to change it if folks have other preferences.

Reply:
Jorge shared the same feedback below, so I addressed it at  https://github.com/WordPress/gutenberg/pull/73642/commits/8d91d5053c53c9c3963c9c1ff191e6a9bc1d57b4